### ISSUE TITLE
CODEBASE 16 doesn't recognize outlook 2021

### ISSUE THREAD

**Issue by udiudiudiudi** [phw198/OutlookGoogleCalendarSync#1332]
when you load the 2.9 version, it shows error report that the codebase 16 doesn't recognize the outlook version

which is 2021

**Comment by udiudiudiudi** [comment]
Hello? do you plan on releasing a version that would support outlook 2021 any time soon?

**Comment by phw198** [comment]
I [already explained](https://github.com/phw198/OutlookGoogleCalendarSync/issues/1312#issuecomment-939338276) this error isn't indicative of OGCS not supporting Outlook 2021. It simply means it wasn't able to determine the full name of the product via its registry entry which, whilst should be resolved, certainly isn't the cause of #1312.

From your logfile on the other issue:
```
2021-10-09 10:45:53,889 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.Factory [111] -  Checking for more accurate Outlook version for codebase v16.
2021-10-09 10:45:53,890 WARN  1 OutlookGoogleCalendarSync.OutlookOgcs.Factory [144] -  Configuration subdirectory does not exist.
2021-10-09 10:45:53,891 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.Factory [147] -  REGISTRY
```
This means your registry `HKLM\SOFTWARE\Microsoft\Office\ClickToRun` does not have a `Configuration` folder or a `ProductReleaseIds` key as would be expected. How "official" was the install?

if you search for a `ProductReleaseIds` key elsewhere in your registry, does one exist and if so, what's the path?

**Comment by udiudiudiudi** [comment]
thats what i have inside ProductReleaseIds

ProPlus2021Retail,ProPlus2021Volume

**Comment by phw198** [comment]
What's the full path?

**Comment by udiudiudiudi** [comment]
HKLM\SOFTWARE\Microsoft\Office\ClickToRun\Configuration\ProductReleaseIds

**Comment by phw198** [comment]
The only subdirectory OGCS can see on your machine is `REGISTRY`, none of the others including `Configuration`. Can you screenshot you registry similar to mine?
![image](https://user-images.githubusercontent.com/11843015/140835874-89eb22d7-1fff-449b-8e97-9b8450768e96.png)

Can you see any difference on the permissions or anything between `REGISTRY` and `Configuration`?

**Comment by jonstelly** [comment]
Seeing same issue, registry screenshot from my machine (I can send you a .reg export file too if needed, but it's a lot of data I'll need to scrub.  If you ask me about specific keys I can get you their values)

![image](https://user-images.githubusercontent.com/967068/140962321-d0f6cf87-2a5e-47bf-a28e-22bae967d928.png)

**Comment by jonstelly** [comment]
PR submitted, just needed an enum for Office 2021 Retail.  Tested and synced on my machine

**Comment by phw198** [comment]
@jonstelly I think your issue is different then? The OP wasn't even able to see the `Configuration` registry subkey, let alone see the product ID values - it [failed here](https://github.com/phw198/OutlookGoogleCalendarSync/blob/92ee4400ebd5baf22014775f9fe0436dcd2f906e/src/OutlookGoogleCalendarSync/OutlookOgcs/OutlookFactory.cs#L124). 

Can you post the logfile lines around the failure you were getting so I can confirm?

**Comment by jonstelly** [comment]
Ahh, my bad.  I did get the same error message on launch but now I see they have 2021 Volume and Retail where I only have Retail.

**Comment by phw198** [comment]
@jonstelly No worries, I'll get the `ProPlus2021Retail` into the next release - but it sounds like you're not having an issue in general with Outlook 2021 being compatible with OGCS? For example, #1312 where new items created in Outlook 2021 are apparently not visible to OGCS?

**Comment by phw198** [comment]
:notebook: Just to note, `ProPlus2021Volume` and `ProPlus2021Retail` were included in v2.9.2
